framework/src/modules/chain/bft/finality_manager.js

```diff
@@ -248,20 +253,16 @@ class FinalityManager extends EventEmitter {
 				) {
 					return needleHeight;
 				}
-
 				// Move the needle to previous block and consider it current for next iteration
-				needleHeight = previousBlockHeader.maxHeightPreviouslyForged;
 				currentBlockHeader = previousBlockHeader;
-			} else {
-				needleHeight -= 1;
 			}
+			needleHeight -= 1;
 		}
 		return needleHeight;
 	}
 
 	recompute() {
 		this.state = {};
-		this.finalizedHeight = this._initialFinalizedHeight;
```

Comment:
Recomputation meant to re-compute everything, including the finalized height. So what's the synopsis for not resetting its value?

Reply:
Finalized height cannot be recomputed in all the case (from the LIP), therefore, we need to store it in the chain state